astilectron.go

```diff
@@ -444,3 +447,35 @@ func (a *Astilectron) NewTray(o *TrayOptions) *Tray {
 func (a *Astilectron) NewNotification(o *NotificationOptions) *Notification {
 	return newNotification(a.worker.Context(), o, a.supported != nil && a.supported.Notification != nil && *a.supported.Notification, a.dispatcher, a.identifier, a.writer)
 }
+
+type eventProxy struct {
+	id         string
+	dispatcher *dispatcher
+	logger     astikit.SeverityLogger
+}
+
+func (g *eventProxy) On(eventName string, l Listener) {
+	g.dispatcher.addListener(g.id, eventName, l)
+}
+
+func (a *Astilectron) ExecuteJavascript(code string, response interface{}) error {
+	ep := &eventProxy{
+		id:         a.identifier.new(),
+		dispatcher: a.dispatcher,
+		logger:     a.l,
+	}
+	event, err := synchronousEvent(
+		a.worker.Context(), ep, a.writer,
+		Event{Name: EventNameAppExecuteJavaScript, TargetID: ep.id, Code: code},
+		EventNameAppExecuteJavaScriptCallback)
+	if err != nil {
+		return fmt.Errorf("sending ExecuteJavascript event failed: %w", err)
+	}
+	if event.Error != nil && *event.Error != "" {
+		return fmt.Errorf("ExecuteJavascript failed: %s", *event.Error)
+	}
+	if err := json.Unmarshal([]byte(event.Reply), response); err != nil {
```

Comment:
1) Could you wrap this in `if response != nil && len(event.Success) > 0` because some developers may not need a response here
2) If you use the `Success` field as I mentioned, you can use `json.Unmarshal(event.Success, response)` instead